src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -304,10 +312,30 @@ public object Instance
             {
                 // We create this lazily because we want the constructor in a workflow context
                 instance ??= Definition.CreateWorkflowInstance(startArgs!.Value);
+                // Dynamic options method needs to be invoked at this point, after initting the
+                // workflow instance but before performing any activations.
+                _ = DynamicDefinitionOptions;
```

Comment:
If you're always running this anyways, then you don't need to make it lazy IMO, can just set the definition options right here. Before I didn't think you cared about when you called the dynamic definition method, but I can see now it is better to be predictable. My concern was what happens if the dynamic options method accesses `Workflow.Instance` per https://github.com/temporalio/sdk-dotnet/pull/466#discussion_r2070621271?

Reply:
Yeah, that was exactly the point of doing it inline there. It needs to happen at the right time